From 36eaa23a32ee6de7e1445835a854ffc40e209d2a Mon Sep 17 00:00:00 2001
From: gokulc <gokulc@ami.com>
Date: Sun, 19 Jul 2026 18:00:20 +0530
Subject: [PATCH] SDKv11.02-fix-migration-for-SPI-Driver 
mtd: spi: Add support for Winbond W25QxxRV serial NOR flash devices
On W25QxxRV parts, the configuration register cannot be programmed
using the 01h (status register program) command. Use the 31h
command instead. If the 01h command is used, the last 8 clock
cycles are ignored.
Signed-off-by: gokulc <gokulc@ami.com>
---
 drivers/mtd/spi/spi-nor-core.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/drivers/mtd/spi/spi-nor-core.c b/drivers/mtd/spi/spi-nor-core.c
index e1936759..513fdd40 100644
--- a/drivers/mtd/spi/spi-nor-core.c
+++ b/drivers/mtd/spi/spi-nor-core.c
@@ -1606,6 +1606,27 @@ static int write_sr_cr(struct spi_nor *nor, u8 *sr_cr)
 		return ret;
 	}
 
+#if defined(CONFIG_SPI_FLASH_WINBOND)
+	if (JEDEC_MFR(nor->info) == SNOR_MFR_WINBOND) {
+		write_enable(nor);
+
+		ret = nor->write_reg(nor, SPINOR_OP_WINBOND_WRSR2, &sr_cr[1], 1);
+		if (ret < 0) {
+			dev_dbg(nor->dev,
+				"error while writing configuration register\n");
+			return -EINVAL;
+		}
+
+		ret = spi_nor_wait_till_ready(nor);
+		if (ret) {
+			dev_dbg(nor->dev,
+				"timeout while writing configuration register\n");
+			return ret;
+		}
+
+		write_disable(nor);
+	}
+#endif
 	return 0;
 }
 
-- 
2.53.0

